glusterlib/backend_reset.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+#
+# Copyright 2015 Nandaja Varma <nvarma@redhat.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Library General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
+
+from yaml_writer import YamlWriter
+from conf_parser import ConfigParseHelpers
+from global_vars import Global
+from helpers import Helpers
+import re
+
+
+class BackendReset(YamlWriter):
+
+    def __init__(self, config):
+        self.config = config
+        section_regexp = '^backend-reset(:)*(.*)'
```

Comment:
Yes, I am checking if there are any sections starting with backend-reset. It can then have a subsection in the format backend-setup:10.70.46.13(host specific). I wanted to avoid an extra stripping step so grouped the colon and the hostname separately. 
answer to question 1: yes this is expected
answer to question 2: The group (.*) if present is the hostname implying hostvars, else it means it is common data for all the hosts under the 'hosts' section implying groupvars.
Hope that makes sense.